sandy/core/cov.py

```diff
@@ -662,18 +657,13 @@ def gls_update(self, Vy, S, delta, threshold=None):
 
         Returns
         -------
-        TYPE
-            DESCRIPTION.
+        `CategoryCov`
+            GLS method apply to a CategoryCov object for a given Vy and S.
 
         """
-        # Check the data entered
-        S_ = pd.Dataframe(S)
-        Vy_ = pd.Series(Vy)
-
-        # Custom GLS:
         Vx = self.data
-        A = self._gls_sensitivity(S_, Vy_)
-        V_new = Vx - A.data.dot(Vx)
+        A = self._gls_sensitivity(S, Vy)

```

Comment:
Why don't you do anymore `pd.DataFrame(S)`?

Reply:
Because it is done in _gls_sensitivity for the two variables